### PR TITLE
change(ci): Create automatic tickets on CI failure for more workflows

### LINKED ISSUE #7678: Automatically open an issue when release jobs fail in CD
https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/continous-delivery.yml
https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/release-binaries.yml

### LINKED ISSUE #7679: Automatically open an issue when crates jobs fail in CI
https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/release-crates-io.yml
https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/continous-integration-os.yml
https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/build-crates-individually.yml

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

In https://github.com/ZcashFoundation/zebra/pull/7565 we added and tested (https://github.com/ZcashFoundation/zebra/pull/7565) to open new tickets or update already created ones when the CI fails in a particular workflow.

This PR use that template to add more jobs of the same kind in additional workflows.

It also do some minor changes to the the original template. 

## Solution

Repeat the `failure-issue` job in the following workflows:

releases:
https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/continous-delivery.yml
https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/release-binaries.yml
main branch merges:
https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/release-crates-io.yml
https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/continous-integration-os.yml
https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/build-crates-individually.yml

The lower priorities of https://github.com/ZcashFoundation/zebra/issues/6864#issue-1746867363 are not added to this pull request.

## Review

This is a draft until test are done. Test will be in the form of a separated pull request where the base is this one, similar to https://github.com/ZcashFoundation/zebra/pull/7569

Apart from that, it can be reviewed.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

File: .github/workflows/continous-integration-docker.yml
Comment (oxarbitrage) [inline .github/workflows/continous-integration-docker.yml:795]: I think we can just do failure AND cancelled.

File: .github/workflows/release-binaries.yml
Comment (oxarbitrage) [inline .github/workflows/release-binaries.yml:52]: We don't have `github.event.pull_request` available here.

**Comment by oxarbitrage** [comment]
Adding "do not merge" tag for the release in process.

TODO: Remove label after the release is done.

File: .github/workflows/release-binaries.yml
Comment (teor2345) [inline .github/workflows/release-binaries.yml]: I think we want to fail if our experimental image fails, because building is reliable. It's only running on testnet that is unreliable, and the image build doesn't do that.

```suggestion
    needs: [ build, build-mining-testnet ]
```

**Comment by oxarbitrage** [comment]
Test was made in https://github.com/ZcashFoundation/zebra/pull/7621

https://github.com/ZcashFoundation/zebra/pull/7621#issuecomment-1738166385

**Comment by mpguerra** [comment]
@gustavovalverde can we have your final review on this one?

**Comment by mpguerra** [comment]
I have split off these 2 items from #6864 into 2 issues that are scheduled for this sprint and will be closed when this PR merges. 

#6864 is now a tracking issue instead.

**Review by teor2345 — APPROVED** [review]
Let's see how this goes!

**Comment by teor2345** [comment]
@mergifyio refresh

**Comment by teor2345** [comment]
Failed in the merge queue in PR #7692 due to issue #7659, I've re-queued it.

**Pull request merged**
